README.md

```diff
@@ -50,12 +50,14 @@ Clicking the "Stop" button in the debugger will kill the Bazel process being
 debugger, allowing you to halt the current build. The Bazel server, however,
 will continue running.
 
-## Using the LSP (Experimental)
+## Using a language server (Experimental)
 
-This extension can use [Facebook's starlark project](https://github.com/facebookexperimental/starlark-rust) as a language server.
+Some of the functionality of this extension, such as go to definition and completions, can be replaced with a language server. There are currently two in-progress implementations of this:
```

Comment:
I wonder, if in the longterm we should aim for also replacing buildifier-based functionality (formatting, linting) with the language servers. As a part-time vim user, I personally would like that, because I can easily use a language server from vim, but I can't directly use buildifier. Also, it would probably simplify this extension a bit, if we have a single dependency (the language server) instead of two dependencies (the language server and buildifier).

(not really something to change in this pull request. Just a general idea)

Reply:
rust-analyzer does something similar where `cargo check` (or `clippy`? forgot which one) is actually run by the language server itself and not the extension iirc

Reply:
yes, `clangd` (for C++) also does something similar: They also embed the `clang-tidy` checks into `clangd` such that they can be displayed directly in the IDE. clangd does not actually call clang-tidy as a separate binary, though, but instead directly links against the clang-tidy library. Not sure how buildifier is built and if it could also be consumed as a library.

Reply:
rust-analyzer does forward the output of `cargo check`, but there is some doubt on whether this is the correct thing to. See https://github.com/rust-lang/rust-analyzer/issues/13437#issuecomment-1381758374.

Reply:
But yeah I would probably advocate for some sort of middle ground where the language servers are the ones responsible for installing `buildifier` and running it on changes, instead of implementing that functionality themselves. It'd guarantee that we won't have any conflicts with any existing workflows that use `buildifier`, e.g. people running it in CI. (Otherwise if the language servers implemented `buildifier`'s functionality themselves, then to avoid said conflicts we'd have to make sure their behaviors match exactly which would be difficult to do).

Reply:
Ooo interesting, I didn't see that thread!

Reply:
I can't comment on the security side, but I like the idea of this extension downloading tools. rust-analyzer used to do that before it came bundled with rust and the user experience was great.

Reply:
I created #371 to track the idea of automatically bootstrapping the tools